snuba/clickhouse/columns.py

```diff
@@ -68,10 +69,31 @@ def flatten(self, name: str) -> Sequence[FlattenedColumn]:
         return [FlattenedColumn(None, name, self)]
 
 
-class Nullable(ColumnType):
+class ColumnTypeWithModifier(ABC, ColumnType):
```

Comment:
This was in another PR in the stack. I just moved it here because it was not needed anymore to check for nullable columns, now we need it to check for materialized columns.

Reply:
Hoping this is still a "temporary" thing (at least as much as one can exist)

Reply:
if you are talking about the modifiers discussion in the other PR, I would say so, modifiers should be attributes, including the existing ones (Nullable, Materialized, WithCodec, WithDefault, etc.)